azure-servicebus/src/main/java/com/microsoft/azure/servicebus/MessageConverter.java

```diff
@@ -134,7 +136,20 @@ else if (body instanceof AmqpSequence)
 		ApplicationProperties applicationProperties = amqpMessage.getApplicationProperties();
 		if(applicationProperties != null)
 		{
-			brokeredMessage.setProperties(applicationProperties.getValue());
+			Map<String, String> convertedProperties = new HashMap<>();
+			for(Map.Entry<String, Object> entry : applicationProperties.getValue().entrySet())
+			{
+				if(entry.getValue() instanceof String)
+				{
+					convertedProperties.put(entry.getKey(), (String)entry.getValue());
+				}
+				else
+				{
+					convertedProperties.put(entry.getKey(), entry.getValue().toString());
```

Comment:
>.toString() [](start = 61, length = 11)

Do we need to convert to string?

Reply:
That was a mistake in 1.0. Properties was defined as a Map<String, String>, which was fixed in 2.0. For 1.0, we still have to maintain that contract, otherwise existing code will break.

---
In reply to: [242698582](https://github.com/Azure/azure-service-bus-java/pull/313#discussion_r242698582) [](ancestors = 242698582)

Reply:
So when merging with 2.0, you'll remove this toString()

---
In reply to: [242722505](https://github.com/Azure/azure-service-bus-java/pull/313#discussion_r242722505) [](ancestors = 242722505,242698582)